src/aws-lambda/segment-personalize-inference-destination/segment-personalize-inference-destination.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: MIT-0
+#
+# segment-personalize-inference-destination.py
+#  
+# This lambda contains sample code that allows you to use Segment real-time events to perform inference
+# against an Amazon Personalize Campaign.  This lambda is not suitable for training events.
+#
+
+import json
+import boto3
+import os
+import dateutil.parser as dp
+import logging
+import analytics  # Segment Python library
+import requests
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+# Check that the Segment env variables are set correctly
+if not 'segment_personas_write_key' in os.environ or os.environ['segment_personas_write_key'] == '':
+    raise Exception('segment_personas_write_key is null or not defined.')
+else:
+    analytics.write_key = os.environ['segment_personas_write_key']
+
+if not 'recommendations_service_url' in os.environ or os.environ['recommendations_service_url'] == '':
+    raise Exception('recommendations_service_url not configured as environment variable')
+else:
+    recommendations_service_url = os.environ['recommendations_service_url']
```

Comment:
Use CloudMap to lookup or better yet pass the Recommendations service ALB since this is very specific to RDS. If you use CloudMap, put that inside `lambda_handler` since the tasks can change in ECS due to redeploys. Lots of examples of how to do this throughout the RDS codebase.